drivers/timer/npcx_itim_timer.c

```diff
@@ -0,0 +1,250 @@
+	ceiling_fraction(ticks * EVT_CYCLES_PER_SEC, \
+			 CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+/* Instance of system and event timers */
+static struct itim64_reg *const sys_tmr = (struct itim64_reg *)
+					DT_INST_REG_ADDR_BY_NAME(0, sys_itim);
+static struct itim32_reg *const evt_tmr = (struct itim32_reg *)
+					DT_INST_REG_ADDR_BY_NAME(0, evt_itim);
+
+static const struct npcx_clk_cfg itim_clk_cfg[] = NPCX_DT_CLK_CFG_ITEMS_LIST(0);
+
+static struct k_spinlock lock;
+/* Announced cycles in system timer before executing z_clock_announce() */
+static uint64_t cyc_sys_announced;
+/* Current target cycles of time-out signal in event timer */
+static uint32_t cyc_evt_timeout;
+
+/* ITIM local inline functions */
+static inline uint64_t npcx_itim_get_sys_cyc64(void)
+{
+	uint32_t cnt64h, cnt64h_check, cnt64l;
+
+	/* Read 64-bit counter value from two 32-bit registers */
+	do {
+		cnt64h_check = sys_tmr->ITCNT64H;
+		cnt64l  = sys_tmr->ITCNT64L;
+		cnt64h = sys_tmr->ITCNT64H;
+	} while (cnt64h != cnt64h_check);
+
+	cnt64h = NPCX_ITIM64_MAX_HALF_CNT - cnt64h;
+	cnt64l = NPCX_ITIM64_MAX_HALF_CNT - cnt64l + 1;
+
+	/* Return current value of 64-bit counter value of system timer */
+	return (((uint64_t)cnt64h) << 32) | cnt64l;
+}
+
+/* ITIM local functions */
+static void npcx_itim_start_evt_tmr_by_tick(int32_t ticks)
+{
+	/*
+	 * Get desired cycles of event timer from the requested ticks which
+	 * round up to next tick boundary.
+	 */
+	if (ticks <= 0) {
+		ticks = 1;
+	}
+
+	if (ticks == K_TICKS_FOREVER) {
+		cyc_evt_timeout = NPCX_ITIM32_MAX_CNT;
+	} else {
+		cyc_evt_timeout = MIN(EVT_CYCLES_FROM_TICKS(ticks),
+				      NPCX_ITIM32_MAX_CNT);
+	}
+	LOG_DBG("ticks %x, cyc_evt_timeout %x", ticks, cyc_evt_timeout);
+
+	/* Disable event timer if needed and wait for it to take effect */
+	if (IS_BIT_SET(evt_tmr->ITCTS, NPCX_ITCTS_ITEN)) {
+		evt_tmr->ITCTS &= ~BIT(NPCX_ITCTS_ITEN);
+		while (IS_BIT_SET(evt_tmr->ITCTS, NPCX_ITCTS_ITEN))
+			;
+	}
+
+	/* Upload counter of event timer */
+	evt_tmr->ITCNT32 = MAX(cyc_evt_timeout - 1, 1);
+
+	/* Enable event timer and wait for it to take effect */
+	evt_tmr->ITCTS |= BIT(NPCX_ITCTS_ITEN);
+	while (!IS_BIT_SET(evt_tmr->ITCTS, NPCX_ITCTS_ITEN))
+		;
+}
+
+static void npcx_itim_evt_isr(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Disable ITIM event module first */
+	evt_tmr->ITCTS &= ~BIT(NPCX_ITCTS_ITEN);
+	/* Clear timeout status for event */
+	evt_tmr->ITCTS |= BIT(NPCX_ITCTS_TO_STS);
+
+	if (!IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
+		/* Enable event timer for ticking and wait to it take effect */
+		evt_tmr->ITCTS |= BIT(NPCX_ITCTS_ITEN);
+		while (!IS_BIT_SET(evt_tmr->ITCTS, NPCX_ITCTS_ITEN))
+			;
+	}
+
+	k_spinlock_key_t key = k_spin_lock(&lock);
+	uint32_t delta_ticks = (uint32_t)((npcx_itim_get_sys_cyc64() -
+				cyc_sys_announced) / SYS_CYCLES_PER_TICK);
+
+	/* Store announced cycles of system timer */
+	cyc_sys_announced = npcx_itim_get_sys_cyc64();
+	k_spin_unlock(&lock, key);
```

Comment:
This could be guarded with IS_ENABLED(CONFIG_TICKLESS_KERNEL) since delta_ticks is unused otherwise.

Reply:
Done.